src/cascadia/TerminalSettingsEditor/Profiles.idl

```diff
@@ -4,13 +4,58 @@
 import "EnumEntry.idl";
 import "MainPage.idl";
 
+#include "ViewModelHelpers.idl.h"
+
 namespace Microsoft.Terminal.Settings.Editor
 {
+    runtimeclass ProfileViewModel : Windows.UI.Xaml.Data.INotifyPropertyChanged
```

Comment:
Am I dumb for thinking there could be an `IProfile` that defines all these, then `Profile` and `ProfileViewModel` implement that? So these could be defined in a single source of truth?

Reply:
My gut tells me the interface would be good _now_, but bad _in the future_. As we implement UI-driven features that diverge from the representation in the settings model (which we _will_, make no mistake), we'll have to add things to this idl manually anyway.

_Already_, this is incorrect. We should move the bindable enum stuff up to the ViewModel because _it's a viewmodel of an enum value_, and make the active values in the ProfileViewModel use EnumEntries for enum fields. Much of the logic that is in the Profiles page CodeBehind should be here.

This is just a start, and I think doing the interface thing would take us on the wrong path. What do you think?

Reply:
I know the redundancy sucks. FromJson, ToJson, Clone, Copy, TSM idl, TSE idl, TSE xaml, IControlSettings, ICoreSettings, IAppearanceConfig. It's awful.

Reply:
That makes sense to me